cdap-data-fabric-tests/src/test/java/co/cask/cdap/data/stream/service/StreamMetaStoreTestBase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.stream.service;
+
+import co.cask.cdap.api.data.stream.StreamSpecification;
+import co.cask.cdap.proto.NamespaceMeta;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMultimap;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test the {@link StreamMetaStore} implementations.
+ */
+public abstract class StreamMetaStoreTestBase {
+
+  protected abstract StreamMetaStore getStreamMetaStore();
+
+  protected abstract void createNamespace(String namespaceId);
+
+  @Test
+  public void testStreamMetastore() throws Exception {
+    StreamMetaStore streamMetaStore = getStreamMetaStore();
+
+    streamMetaStore.addStream("foo", "bar");
```

Comment:
Hmmm, what does this suggest? Since you have not created the namespace *foo* yet, but are creating a stream in it?
Particularly stands out because in lines 46 and 47 below you create namespaces before creating streams.